app/controllers/lti_deployments_controller.rb

```diff
@@ -14,8 +14,10 @@ def launch
       return
     end
     nonce = rand(10 ** 30).to_s.rjust(30, '0')
-    session[:client_id] = params[:client_id]
-    session[:nonce] = nonce # Nonce should be present in the JWT sent to MarkUs, and must be verified
+    cookies.permanent.encrypted[:client_id] = { value: params[:client_id], expires: 1.hour.from_now }
```

Comment:
Can all of these values be stored in a single cookie and the value be a json string containing all the relevant values? That way we only have one additional cookie to keep track of and we lessen the risk of leaving cookies undeleted.